model/task_queue_service_dependency.go

```diff
@@ -362,6 +362,61 @@ func (d *basicCachedDAGDispatcherImpl) FindNextTask(ctx context.Context, spec Ta
 				}
 			}
 
+			taskVersion, err := VersionFindOneId(nextTaskFromDB.Version)
+			if err != nil {
+				grip.Warning(message.WrapError(err, message.Fields{
+					"dispatcher": DAGDispatcher,
+					"function":   "FindNextTask",
+					"message":    "problem finding version for task in db",
+					"version_id": nextTaskFromDB.Version,
+					"task_id":    nextTaskFromDB.Id,
+					"distro_id":  d.distroID,
+				}))
+				return nil
+			}
+			if taskVersion == nil {
+				grip.Warning(message.Fields{
+					"dispatcher": DAGDispatcher,
+					"function":   "FindNextTask",
+					"message":    "version for task from db not found",
+					"task_id":    nextTaskFromDB.Id,
+					"version_id": nextTaskFromDB.Version,
+					"distro_id":  d.distroID,
+				})
+				return nil
```

Comment:
Would this cause issues if a single host task group can't run its next task? If so, maybe we can make an exception for single host task groups, or somehow unschedule the later task group tasks?

Writing a test for how single host task groups interact with this could also be helpful to understand how limits interact with them!

Reply:
I think that's valid. I think it's best if in-progress task groups are allowed to complete regardless of the limit, however task groups that have yet to start should still abide by the limits. Added a bit more logic + tests for this.